cpp/arcticdb/column_store/column.hpp

```diff
@@ -708,8 +733,49 @@ class Column {
                           functor&& f) {
         auto left_input_data = left_input_column.data();
         auto right_input_data = right_input_column.data();
+        initialise_output_column(left_input_column, right_input_column, output_column);
         auto output_data = output_column.data();
-        std::transform(left_input_data.cbegin<left_input_tdt>(), left_input_data.cend<left_input_tdt>(), right_input_data.cbegin<right_input_tdt>(), output_data.begin<output_tdt>(), std::forward<functor>(f));
+        auto output_it = output_data.begin<output_tdt, false, false>();
+
+        if (!left_input_column.is_sparse() && !right_input_column.is_sparse()) {
+            // Both dense, use std::transform over the shorter column to avoid going out-of-bounds
+            if (left_input_column.row_count() <= right_input_column.row_count()) {
+                std::transform(left_input_data.cbegin<left_input_tdt, false>(), left_input_data.cend<left_input_tdt, false>(), right_input_data.cbegin<right_input_tdt, false>(), output_it, [f = std::forward<functor>(f)](auto left_value, auto right_value) {
+                    return f(left_value, right_value);
+                });
```

Comment:
Can't we use this instead?
```suggestion
                std::transform(left_input_data.cbegin<left_input_tdt, false>(), left_input_data.cend<left_input_tdt, false>(), right_input_data.cbegin<right_input_tdt, false>(), output_it, std::forward<functor>(f));
```